### DISCUSSION TITLE
"Expansions" as tradable NFTs

### DISCUSSION THREAD

**Discussion by brian-man** [sunflower-land/sunflower-land#2020] [General]
### Overview
My recollection is that the 6x6 segments of land are called "Expansions", even though the first three are present at birth of an island so they aren't really expansions in the sense of "extensions of lands you've paid for directly".  Moving on...

According to the docs, each expansion is "owned" (in the blockchain sense) by a farm as an ERC721 token.  My recollection (again) is that these are intended to be -- to some degree -- tradable on external (to the game) blockchain marketplaces.

### Data
The expansion RNG algorithm isn't publicly documented and the source isn't open.  So data mining and estimation are all the players have available currently to work with.

Inu is the most active-on-discord record-keeper (to my knowledge) on the stats on expansions across the active islands on the blockchain, and can likely offer much better (and accurate) data and insight into the current expansion model.

### Categories
It seems to me we have a few categories of expansions currently:
- 1-3: The initial 3 expansions are intended to start a player off and are less "RNG varied".  They include a 3x3 plot grid that is never offered for subsequent expansions, for example.
- 4-9: There is significant variety in these such that there is a huge chasm between the "best" and "worst"  islands that contain 9x6x6 squares.
- 10-13: Fruit patches were introduced here.  The variance for the full set of 4 seems to be in the [2,6] fruit range.
- 14: Rare mine is introduced here and the range seems to be [1,2].
- 15: There seems to be a fairness/balancing effort in this expansion as the previously "worst" farms are getting lots of plots -- up to 8! -- while the previously "best" farms are getting zero plots.  Also the trees appear to be getting balanced to "everybody gets 6" at this point.

### Concern
I haven't heard much about this recently and I really feel like it needs attention (and public discussion) so that a bunch of "undesirable results" (some of them arguably exploits) don't occur on mainnet.

I don't know what the future holds for 16+ (will 16-18 be continued rebalancing?), but I'm nervous that the team isn't putting enough human resources (as evidenced by a lack of public discussion on the topic) into planning this out.

### Examples
The last three categories -- fruit, rare mine, and balancing -- are the most concerning to me.

Some examples follow of player min-maxing in obvious ways.  The team may consider one or more of these (or related activities) to be exploits.

#### Fruitopia
1. Acquire new island
2. Buy and deposit a number of "expansion 10-13" NFTs that each contain 2 fruit patches

#### Rare mine field
1. Acquire new island
3. Buy and deposit a number of "expansion 14" NFTs

#### Exploiting balancing
1. Acquire the worst island you can (via open sea or growing it) where worst means the least plots and fruit trees.
2. Withdraw expansions 15+
4. Expand 15 again => really good "roll" because of balancing
5. Withdraw expansion 15 and sell it
6. Repeat steps 3-4

### Strawman

At this point the "right" answer might be something like:
1. repeat the model used for expansion 15 a few more times until you've levelled the playing field
2. introduce "generic RNG" (that is not sensitive in any way to an island's prior expansions) for subsequent expansions
3. only allow withdraw/deposit/trade of expansions produced by 2

**Comment by Kay1797** [comment]
this is very interesting.
i hope the team will at least look this over with some consideration.

**Comment by cryptokohai** [comment]
How will this reconcile with the fact that different lands cost different resources to mint and that resources on the land cannot be traded but the land itself can?
